src/modding/ModManager.cs

```diff
@@ -784,18 +1170,40 @@ private void ApplyChanges()
             return;
         }
 
+        var checkResult = ModLoader.IsValidModList(enabledMods);
+        if (checkResult.ErrorType >= 0)
```

Comment:
You should create an extension method to check "IsSuccessful" method that could be here rather than relying on the exact integer values.